tests/e2e/network/service_annotations.go

```diff
@@ -315,3 +346,57 @@ func validateInternalLoadBalancer(c clientset.Interface, ns string, url string)
 	utils.Logf("validation finished")
 	return err
 }
+
+func validateLoadBalancerBackendPools(vmssname string, cs clientset.Interface, labels map[string]string, ns string, ports []v1.ServicePort, resourcegrpverify string) {
+	// create deployment
+	serviceName := "validate-lb-mode"
+	lbModeLoadBalancerName := serviceName + string(uuid.NewUUID())
+
+	utils.Logf("Creating deployment " + lbModeLoadBalancerName)
+	deployment := createNginxDeploymentManifest(lbModeLoadBalancerName, labels)
+
+	// create service annotation mode for LoadBalancer
+	mode_name := vmssname
+	annotation := map[string]string{
+		azure.ServiceAnnotationLoadBalancerMode: mode_name,
+	}
+	service := createLoadBalancerServiceManifest(cs, lbModeLoadBalancerName, annotation, labels, ns, ports)
+	_, er := cs.CoreV1().Services(ns).Create(service)
+	Expect(er).NotTo(HaveOccurred())
+	utils.Logf("Successfully created LoadBalancer service " + lbModeLoadBalancerName + " in namespace " + ns)
+
+	By("Waiting for service exposure")
+	pip, err := utils.WaitServiceExposure(cs, ns, serviceName)
+	Expect(err).NotTo(HaveOccurred())
+	
+	iplist, er := utils.ListPublicIPs(cs, resourcegrpverify)
+	Expect(er).NotTo(HaveOccurred())
+	for _, ip := range iplist {
+		if ip.Key == pip {
+			utils.Logf("Public Ip verified")
+		}
+	} //Get loadBalancer Name and resource group name
+	lbNameRE := regexp.MustCompile(`^/subscriptions/(?:.*)/resourceGroups/(.+)/providers/Microsoft.Network/publicIPAddresses/(.+)`)
+	match := lbNameRE.FindStringSubmatch(pip)
+	if len(match) != 3 {
+		fmt.Println("Report error")
+		return
+	}
+	loadBalancername := match[1]
+	pipResourceGroup := match[2]
+	//Get backendpools list
+	backendpools, err := network.ListComplete(cs, pipResourceGroup, loadBalancername)
+	Expect(err).NotTo(HaveOccurred())
+	//check whether vmss name equals the backendpool name
+	counter := 0
+	for _, backendpool := range backendpools {
+		re := regexp.MustCompile(vmssname)
```

Comment:
why use vmssname here? I think you mean to use another regexp here?

Reply:
Yes sir I used another regexp here else I can shift to the traditional approach of dealing with strings